lib/objects/objectsInMemFileDB.js

```diff
@@ -98,8 +101,29 @@ class ObjectsInMemoryFileDB extends InMemoryFileDB {
         }
     }
 
+    /**
+     * Checks if given Id is a meta object, else throws error
+     *
+     * @param {string} id to check
+     * @throws Error if id is invalid
+     */
+    async validateMetaObject(id) {
+        if (this.metaCache[id] === undefined) {
+            // if not cached -> getObject
+            const obj = await this.getObjectAsync(id);
+            if (obj && obj.type === 'meta') {
+                this.metaCache[id] = true;
+            } else {
+                this.metaCache[id] = false;
+                return Promise.reject(`${id} is not an object of type "meta"`);
+            }
+        } else if (this.metaCache[id] === false) {
+            return Promise.reject(`${id} is not an object of type "meta"`);
```

Comment:
Rejecting Promises with strings is an anti-pattern, because you never know if the caught exception is an Error or a string. Also, with strings you don't get any stacktrace for where the error comes from. Better to just
```js
throw new Error(`${id} is not an object of type "meta"`)
```
but the calling code needs to be updated to handle errors instead of strings so it does not print out the entire stack trace in a log message (line 452 and similar).

Reply:
@Apollon77 what's your opinion on Errors vs strings?

Reply:
I like clean way (so Errors), but ioBroker uses both well mixed :-( In fact a normal "print(new Error("..."))" will not print stack trace by default and that sway it was never a topic ... All checks chec for "if (err)" so also here unimportant ...

I have no problem in controller to have strings

Reply:
I've added this to eslint and changed it in the PR.